### PR TITLE
port pvl_adrinverter

### PR DESCRIPTION
**Pull request by VolkrB**
#160  In short:

- Ported mat -> csv adrinverter data file

pvsystem.py
- included import of csv in pvsystem.py retrieve_sam function
- added adrinverter function (two lines are a bit ugly, because i am storing coefficients in a string instead of making a column for each)

modelchain.py
- added adrinverter function to wrapper

test_modelchain.py
- removed xfail for adrinverter
- added tests using csv data (Zigor inverter)
(had to also change the single diode test, since i replaced cecinverter with adrinverter)

test_pvsystem.py
- added test for adrinverter function

#### Need to document after everything looks good

I've compared inverter data from the adrinverter function to the sandiainverter function to see if, at least, the results are of the same magnitude and follow a similar trend. They do, however would be nice to see results from the matlab version to test against.

let me know what i can change/improve etc

### CODE REVIEW

File: pvlib/pvsystem.py [hunk 1]
```diff
@@ -1122,6 +1125,9 @@ def retrieve_sam(name=None, path=None):
         elif name == 'sandiamod':
             csvdata = os.path.join(
                 data_path, 'sam-library-sandia-modules-2015-6-30.csv')
+        elif name == 'adrinverter':
+            csvdata = os.path.join(
```
Comment (wholmgren) [inline pvlib/pvsystem.py]: looks like this might fit on a single line that's less than 80 characters.

File: pvlib/pvsystem.py [hunk 2]
```diff
@@ -2037,6 +2043,37 @@ def snlinverter(v_dc, p_dc, inverter):
     return ac_power
 
 
+def adrinverter(v_dc, p_dc, inverter):
+
+    ce_clean = [','.join(inverter['ADRCoefficients'][1:-1].split())]
+    ce_list = np.array([float(s) for s in ce_clean[0].split(',')])
```
Comment (wholmgren) [inline pvlib/pvsystem.py]: ``inverter['ADRCoefficients']`` should be parsed at import time

File: pvlib/test/test_modelchain.py [hunk 3]
```diff
@@ -33,8 +33,8 @@ def cec_dc_snl_ac_system(sam_data):
     module_parameters['b'] = 0.05
     module_parameters['EgRef'] = 1.121
     module_parameters['dEgdT'] = -0.0002677
-    inverters = sam_data['cecinverter']
-    inverter = inverters['ABB__MICRO_0_25_I_OUTD_US_208_208V__CEC_2014_'].copy()
+    inverters = sam_data['adrinverter']
```
Comment (wholmgren) [inline pvlib/test/test_modelchain.py]: maybe I'm forgetting something, but don't we want a new test here instead of changing the existing test?

File: pvlib/data/adr-library.csv [hunk 4]
```diff
@@ -0,0 +1,1762 @@
+Name,Manufacturer,Model,Source,Vac,Vintage,Pacmax,Pnom,Vnom,Vmin,Vmax,ADRCoefficients,Pnt,Vdcmax,Idcmax,MPPTLow,MPPTHi,TambLow,TambHi,Weight
+0,0,0,0,0.0,0.0,0.0,0,0,0,0,0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
```
Comment (wholmgren) [inline pvlib/data/adr-library.csv]: are these 2 lines of 0s here because of the ``_parse_raw_sam_df`` function's ``skip_rows`` argument? I think we should keep this file clean and make the pvlib parser smarter. I don't remember the logic for that parser (I don't even remember if I wrote it).

File: pvlib/test/test_modelchain.py (hunk 3 above)
Comment (wholmgren) [inline pvlib/test/test_modelchain.py]: on a second read, maybe this is ok because all of the methods are still being tested in some way. Is that right?

File: pvlib/pvsystem.py (hunk 1 above)
Comment (VolkrB) [inline pvlib/pvsystem.py]: sure

File: pvlib/pvsystem.py (hunk 2 above)
Comment (VolkrB) [inline pvlib/pvsystem.py]: Yea, seems like that's where it should be

File: pvlib/test/test_modelchain.py (hunk 3 above)
Comment (VolkrB) [inline pvlib/test/test_modelchain.py]: Yea they are, I think the coverage is the same- this was something I was debating as well.

File: pvlib/data/adr-library.csv (hunk 4 above)
Comment (VolkrB) [inline pvlib/data/adr-library.csv]: Yea exactly. I could write a conditional statement in ```_parse_raw_sam_df``` to only skip rows for certain csvs, but maybe you can think of something better than that?

Comment (wholmgren) [inline pvlib/data/adr-library.csv]: I'm struggling to think of a good solution for this and the array parsing issue. One relatively simple option is to replace the 0 lines with metadata lines like in the sam files. So, we'd want a units line and then I guess a copy of the column names. Or maybe just empty cells instead of 0s would be better? The end of the parser function could have a conditional on the presence of 'ADRCoefficients' for the array parsing issue discussed above.

File: pvlib/test/test_modelchain.py (hunk 3 above)
Comment (wholmgren) [inline pvlib/test/test_modelchain.py]: ok with me to leave it alone then

**Comment by VolkrB** [comment]
Updated the csv and moved ```inverter['ADRCoefficients']``` parsing to ```pvsystem._parse_raw_sam_df```

File: pvlib/pvsystem.py [hunk 5]
```diff
@@ -1177,6 +1176,15 @@ def _parse_raw_sam_df(csvdata):
 
     df.index = parsedindex
     df = df.transpose()
+    if 'ADRCoefficients' in df.index:
+        ad_ce = 'ADRCoefficients'
+        df.loc[ad_ce] = df.loc[ad_ce].map(lambda x:
+                                          np.array([float(s) for s in
+                                                    [','.join(x[1:-1]
+                                                              .split())]
+                                                    [0].split(',')]))
```
Comment (VolkrB) [inline pvlib/pvsystem.py]: I realize this is a pretty hackie, ugly, line of code. Trying to see if I can get this to be more elegant.

Comment (wholmgren) [inline pvlib/pvsystem.py]: I think this works: lambda x: list(map(float, x.strip(' []').split(' ')))

not sure if there are pros/cons to adding np.array around the list. 

also, you don't have to stay justified with the opening parenthesis. you can make only a 4 space indent on the line below the opening parenthesis. that alone would help readability!

I'd get rid of the empty else block, too.

other than that, I think you only need to add documentation and then we'll be set.

File: pvlib/pvsystem.py [hunk 6]
```diff
@@ -2037,6 +2043,37 @@ def snlinverter(v_dc, p_dc, inverter):
...
+    p_nt = inverter['Pnt']
+
+    pdc = p_dc/p_nom
+    vdc = v_dc/v_nom
+    poly = np.array([pdc**0, pdc, pdc**2, vdc-1, pdc*(vdc-1),
+                     pdc**2*(vdc-1), 1/vdc-1, pdc*(1./vdc-1),
+                     pdc**2*(1./vdc-1)])
+
+    p_loss = np.dot(ce_list, poly)
+    ac_power = p_nom * (pdc-p_loss)
```
Comment (adriesse) [inline pvlib/pvsystem.py]: the name "ac_power" seems to be breaking the naming pattern...

File: pvlib/pvsystem.py (hunk 5 above)
Comment (adriesse) [inline pvlib/pvsystem.py]: Maybe a code comment should go here to explain what this line is supposed to do.  Does is really all have to be done in one line?

**Comment by adriesse** [comment]
I reviewed the code and didn't see any mistakes.  Since all the coefficients are normalized, it would pretty easy to create some synthetic test cases.  For example, each coefficient with a value of 4% produces a loss of Pnom * 4% at Pdc==Pnom and Vdc==Vnom.  Depending on which particular coefficient it is, this loss will vary with Pdc, Vdc, both, or neither.

**Comment by VolkrB** [comment]
I cleaned up that big ugly parsing line based on @wholmgren suggestions. 

@adriesse: I always struggle with variable naming, so perhaps you, @wholmgren  and I can all agree on something. Also, could you review my pushed documentation and suggest proper definition, references, etc? (i guess this would also be a good place to discuss the naming of vars)

**Comment by adriesse** [comment]
The shorter parsing line is definitely easier to parse, but it still wouldn't hurt to throw a comment in there which would be even easier to parse for readers.

I see that the ac_power name came from the other inverter function and is used in matlab.  I was anticipating something like pac or p_ac, but there doesn't seem to be a risk of misunderstanding so I don't mind if you leave it.

Can you just copy the documentation from the matlab version?

**Comment by VolkrB** [comment]
Updated docs/whatsnew. Let me know if I need to add/change etc.

File: pvlib/pvsystem.py [hunk 7]
```diff
@@ -1059,6 +1059,7 @@ def retrieve_sam(name=None, path=None):
         * CEC module database
         * Sandia Module database
         * CEC Inverter database
+        * Antone Driesse Inverter database
```
Comment (wholmgren) [inline pvlib/pvsystem.py]: Anton

File: pvlib/pvsystem.py [hunk 8]
```diff
@@ -2048,30 +2051,40 @@ def snlinverter(v_dc, p_dc, inverter):
 
 def adrinverter(v_dc, p_dc, inverter):
     r'''
-    Converts DC power and voltage to AC power using Sandia's
-    Grid-Connected PV Inverter model.
+    PVL_ADRINVERTER Converts DC power and voltage to AC power
```
Comment (wholmgren) [inline pvlib/pvsystem.py]: no PVL_ADRINVERTER

File: pvlib/pvsystem.py [hunk 9]
```diff
@@ -2048,30 +2051,40 @@ def snlinverter(v_dc, p_dc, inverter):
...
     p_dc : numeric
-        A scalar or DataFrame of DC powers, in watts, which are provided
-        as input to the inverter. Pdc must be >= 0.
+         A scalar or pandas series of DC powers, in watts, which are provided
+         as input to the inverter. If Vdc and Pdc are vectors, they must be
+         of the same size. p_dc must be >= 0. (W)
 
     inverter : dict-like
-        A dict-like object defining the inverter to be used.
+        A struct defining the inverter to be used, giving the
```
Comment (wholmgren) [inline pvlib/pvsystem.py]: dict-like

File: pvlib/pvsystem.py [hunk 10]
```diff
@@ -2048,30 +2051,40 @@ def snlinverter(v_dc, p_dc, inverter):
...
+        inverter performance parameters according to the model
+        developed by Anton Driesse[1].
```
Comment (wholmgren) [inline pvlib/pvsystem.py]: space before [1]

File: pvlib/pvsystem.py [hunk 11]
```diff
@@ -2048,30 +2051,40 @@ def snlinverter(v_dc, p_dc, inverter):
...
         A set of inverter performance parameters are provided with
-        pvlib, or may be generated from the library using retrievesam. 
+        pvlib, or may be generated from the library using retrievesam.
```
Comment (wholmgren) [inline pvlib/pvsystem.py]: maybe something like "loaded from the supplied data table using retrieve_sam" would be more accurate.

File: pvlib/pvsystem.py [hunk 12]
```diff
@@ -2048,30 +2051,40 @@ def snlinverter(v_dc, p_dc, inverter):
...
         See Notes for required keys.
 
     Returns
     -------
     ac_power : numeric
-        Modeled AC power output given the input DC voltage, v_dc, and
-        input DC power, p_dc.
+        A numpy array of modeled AC power output given the input
```
Comment (wholmgren) [inline pvlib/pvsystem.py]: "A numpy array" is not quite right. It returns a Series if the input data is also a Series.

File: pvlib/pvsystem.py [hunk 13]
```diff
@@ -1178,7 +1180,8 @@ def _parse_raw_sam_df(csvdata):
     df = df.transpose()
     if 'ADRCoefficients' in df.index:
         ad_ce = 'ADRCoefficients'
-        df.loc[ad_ce] = df.loc[ad_ce].map(lambda x:list(
+        # parses string into list containing floats:
```
Comment (wholmgren) [inline pvlib/pvsystem.py]:
```
# for each inverter, parses a string of coefficients like
# ***insert example***
# into a list containing floats
```

File: pvlib/pvsystem.py (hunk 7 above)
Comment (VolkrB) [inline pvlib/pvsystem.py]: (Sorry Anton)

File: pvlib/pvsystem.py [hunk 14]
```diff
@@ -1122,6 +1127,8 @@ def retrieve_sam(name=None, path=None):
         elif name == 'sandiamod':
             csvdata = os.path.join(
                 data_path, 'sam-library-sandia-modules-2015-6-30.csv')
+        elif name == 'adrinverter':
+            csvdata = os.path.join(data_path, 'adr-library.csv')
```
Comment (adriesse) [inline pvlib/pvsystem.py]: The parameters in this particular file were generated using the CEC efficiency measurements available to me as of 2013-10.  It might be good to add the date suffix.  I'm looking into doing an update.

File: pvlib/pvsystem.py [hunk 15]
```diff
@@ -2037,6 +2051,108 @@ def snlinverter(v_dc, p_dc, inverter):
...
+    pdc = p_dc/p_nom
+    vdc = v_dc/v_nom
+    poly = np.array([pdc**0, pdc, pdc**2, vdc-1, pdc*(vdc-1),
+                     pdc**2*(vdc-1), 1/vdc-1, pdc*(1./vdc-1),
+                     pdc**2*(1./vdc-1)])
+
+    p_loss = np.dot(np.array(ce_list), poly)
+    ac_power = p_nom * (pdc-p_loss)
+
+    p_nt = -1*np.absolute(p_nt)
```
Comment (adriesse) [inline pvlib/pvsystem.py]: Earlier versions of the cec/sam/sandia database were not consistent: most inverters had positive p_nt but many were negative.  The latest sam database I looked at had only positve values.

File: pvlib/pvsystem.py (hunk 14 above)
Comment (VolkrB) [inline pvlib/pvsystem.py]: ok I will add a date suffix

File: pvlib/pvsystem.py (hunk 15 above)
Comment (VolkrB) [inline pvlib/pvsystem.py]: so ```p_nt = -1*np.absolute(p_nt)``` should just be ```-1*p_nt``` ?

Comment (adriesse) [inline pvlib/pvsystem.py]: That was just background info.  The absolute() shouldn't be needed, but makes it more robust.

**Comment by VolkrB** [comment]
I ran flake8 and added a date suffix

**Comment by wholmgren** [comment]
Looks good to me. We can ignore the failing travis builds. 

@adriesse -- you get to make the final stamp of approval since your name is on it!

File: pvlib/test/test_pvsystem.py [hunk 16]
```diff
@@ -577,6 +578,16 @@ def test_snlinverter(sam_data):
...
+def test_adrinverter(sam_data):
+    inverters = sam_data['adrinverter']
+    testinv = 'Zigor__Sunzet_3_TL_US_240V__CEC_2011_'
+    vdcs = pd.Series(np.linspace(0,50,3))
+    idcs = pd.Series(np.linspace(0,11,3))
+    pdcs = idcs * vdcs
+
+    pacs = pvsystem.adrinverter(vdcs, pdcs, inverters[testinv])
+    assert_series_equal(pacs, pd.Series([-0.25000, 105.635043, 503.769061]))
+
```
Comment (adriesse) [inline pvlib/test/test_pvsystem.py]: This test may catch some types of future bugs, but it's not a good PV example.  The current, voltage and power ranges are way too small for this Zigor model inverter.  

Another idea might be to use some voltage and current combinations found in the pdf file on the CEC/gosolar website and look for the results of the model to be approxiamtely equal.

**Comment by adriesse** [comment]
Other than the test everything looks good to me.

**Comment by VolkrB** [comment]
I don't mind updating the test. I've found a nice reference on gosolar and working on it now

**Comment by VolkrB** [comment]
@adriesse could you comment on the changes to the test when you get a chance?

**Comment by adriesse** [comment]
Sure.  So this is indeed a more realistic set of test conditions.  Did you calculate the expected outputs using an independent method, e.g. Excel?

Unfortunately you dropped the case that tests the night-time tare functionality and this got me thinking that the clipping condition isn't tested either.  I guess this wouldn't show up in the code coverage metrics since it is done with a "where" rather than "if", but it still seems like a good idea to include them.

A test case with scalar inputs, like for pvl_snlinverter, would be good as well.

We could stop here, or we could go further--but that might lead to changes to the inverter code.  What should happen when Pdc or Vdc is less than zero?  There are in fact many realistic and non-realistic combinations of Pdc and Vdc where the model would not produce a realistic output, i.e. it should not be used to extrapolate beyond the normal operating range of the inverter. 

What do you think about producing nan's for out-of-range conditions?  This behavior could be triggered by an optional parameter "range_check", or even multiple parameters for different checks.  Another option would be to return one of more columns of flags.

**Comment by VolkrB** [comment]
"Sure. So this is indeed a more realistic set of test conditions. Did you calculate the expected outputs using an independent method, e.g. Excel?"
- I used/compared values from gosolarcalifornia.

"Unfortunately you dropped the case that tests the night-time tare functionality and this got me thinking that the clipping condition isn't tested either. I guess this wouldn't show up in the code coverage metrics since it is done with a "where" rather than "if", but it still seems like a good idea to include them."
- Seems like I would just have to add an upper and lower limit to the pd.Series.

"What do you think about producing nan's for out-of-range conditions? This behavior could be triggered by an optional parameter "range_check", or even multiple parameters for different checks. Another option would be to return one of more columns of flags."
- I'm not opposed to that. Should the snlinverter have something like this as well?

**Comment by cwhanse** [comment]
> What do you think about producing nan's for out-of-range conditions? This behavior could be triggered by an optional parameter "range_check", or even multiple parameters for different checks. Another option would be to return one of more columns of flags.

I think nan should be reserved for combinations of inputs where the model equation fails to produce a real value, or produces a non-physical result, such as negative power below the night tare.  If the model equation operates and produces meaningless results, I'd rather let the results themselves communicate that failing.

We could consider building an array check function, like is implemented in SAM, that compares the string voltage to the inverter input range, etc., and gives a warning when the array specification appears wrong.

**Comment by wholmgren** [comment]
Definitely return nan if the output is non-physical. Otherwise it's up to model to define good limiting behavior and it's up to the user to not do something dumb. 

Part of the problem with warnings is that we need to be consistent throughout the library. Users see a warning about something in one function and then they expect that they'll be warned about bad behavior everywhere. It's so much easier to be consistently silent.

**Comment by adriesse** [comment]
Ok, then let's get physical.  Inverters simply cannot operate beyond a certain input voltage range nor inject power into a PV array.  The reasons vary with topology and components used.  I would propose (untested):
```
def adrinverter(v_dc, p_dc, inverter, vtol=0.10):
...
v_limit_upper = np.nanmax([Vmax, Vdcmax, MPPTHi]) * (1 + vtol)
v_limit_lower = np.nanmin([Vmin, MPPTLow]) * (1 - vtol)

ac_power = np.where(v_dc > v_limit_upper, np.nan, ac_power)
ac_power = np.where(v_dc < v_limit_lower, np.nan, ac_power)

ac_power = np.where(p_dc < 0.0, np.nan, ac_power)
```

Regarding the tests:

> "Sure. So this is indeed a more realistic set of test conditions. Did you calculate the expected outputs using an independent method, e.g. Excel?"
> 
>     I used/compared values from gosolarcalifornia.
> 
I'm not sure you we understood each other here.  Where did the values [1161.5745, 1116.4459, 382.6679] come from?

> "Unfortunately you dropped the case that tests the night-time tare functionality and this got me thinking that the clipping condition isn't tested either. I guess this wouldn't show up in the code coverage metrics since it is done with a "where" rather than "if", but it still seems like a good idea to include them."
> 
>     Seems like I would just have to add an upper and lower limit to the pd.Series.
> 
I'm not sure exactly what you mean by that, but it could be right.

I also meant to say earlier that I don't see why the tests are defined with voltage and _current_, which are then multiplied to get power.  Current isn't seen explicitly in the original data or in the model code, so I suggest specifying the power levels directly.

**Comment by VolkrB** [comment]
> I'm not sure you we understood each other here. Where did the values [1161.5745, 1116.4459, 382.6679] come from?

http://www.gosolarcalifornia.ca.gov/equipment/inverter_tests/summaries/Ablerex%20ES%202200-US-240.pdf I picked several input voltages on this doc and then compared the output efficiency to the code's output i.e p_dc/p_ac. They agreed within 1%. I guess I could be more precise in my testing, perhaps someone could check if the numbers agree with the matlab version? 

> I'm not sure exactly what you mean by that, but it could be right.

I mean 1A, 0W as the lower bound and something like 1A, 2500W as the upper. That should take care of tare and clipping conditions?

**Comment by VolkrB** [comment]
Made changes based on input from @adriesse

File: pvlib/pvsystem.py [hunk 17]
```diff
@@ -2051,7 +2051,7 @@ def snlinverter(v_dc, p_dc, inverter):
     return ac_power
 
 
-def adrinverter(v_dc, p_dc, inverter):
+def adrinverter(v_dc, p_dc, inverter, vtol=0.10):
```
Comment (adriesse) [inline pvlib/pvsystem.py]: vtol : numeric
A unit-less fraction that determines how far the efficiency model is allowed to extrapolate beyond the inverter's normal input voltage operating range.

Comment (VolkrB) [inline pvlib/pvsystem.py]: updated

**Comment by adriesse** [comment]
I think this is good, Volker.  With the added comment for the optional parameter I think we could stop there.

**Comment by adriesse** [comment]
I wonder how many people are actually watching this thread...

**Comment by jforbess** [comment]
I'm reading it. I get all the threads in my email, and read them at some
regular frequency.

**Comment by wholmgren** [comment]
It's not immediately clear to me if/how the vtol parameter is being tested. Maybe a comment in the test function would be enough. We do need at least one more test that sets the parameter, though.

**Comment by adriesse** [comment]
There is one case with v_dc below the threshold determined by vtol, and another with v_dc above.  These both produce nan outputs.

I can image tons of additional tests, and garbage in/out scenarios but don't have a good feel for what's appropriate in this context.

**Comment by wholmgren** [comment]
Thanks for clarifying, Anton. In that case I recommend one additional test that uses the same input but with vtol set sufficiently large so that the returned values are no longer nan. This way we have a test that actually sets the value of the parameter instead of relying on the default value, thus testing both the presence of the keyword and that its value is respected.

**Comment by adriesse** [comment]
That seems a bit like testing the compiler's handling of parameter defaults, doesn't it?!

I have another suggestion. How about putting the inverter parameters in the code rather than reading them from the file?  This removes a dependency and also makes clear to the reader what the power and voltage ranges are for the test.

**Comment by wholmgren** [comment]
My primary concern is testing that the function can be called when the keyword is used. This ensures that a test will fail if anyone ever tries to remove ``vtol`` or change ``vtol`` to something else. 

Regarding testing the value of the keyword, I am not at all concerned about the interpreter, but rather I am concerned that the value is actually used later in the code. For example, in prototyping the function there could have been a hardcoded ``vtol = 0.1`` line that was not removed. You would be right to say it should be obvious when reviewing the code, but I make these kinds of mistakes all the time. I am pretty sure that this has happened in pvlib python before. I think it was in singlediode, desoto, or maybe the earlier version of the sapm function.

I like the idea of hardcoding the test parameters for the ``adrinverter`` function. That makes me realize that we're not explicitly testing the ``retrieve_sam`` function for any of the data types. For this PR, we'd only need to worry about testing it with the 'adrinverter' option.

**Comment by wholmgren** [comment]
Hi @VolkrB, can you merge/rebase your PR and add one more simple test using ``vtol`` as described above? Any test that puts ``vtol`` in the function call would be fine with me. Testing using hardcoded values and testing ``retrieve_sam`` can be implemented later. Thanks!

**Comment by VolkrB** [comment]
Sorry I haven't looked at this in a while. I hope the merge/push looks ok

**Comment by wholmgren** [comment]
There are some errors in the test suite that are related to the combination of changing the default ModelChain inverter and the introduction of the vtol/nans.

**Comment by VolkrB** [comment]
oops. Seems like that test has non-physical results. I put in the appropriate nans in the params

File: pvlib/test/test_modelchain.py (hunk 3 above)
Comment (wholmgren) [inline pvlib/test/test_modelchain.py]: I'm only now realizing that this change to the ``cec_dc_snl_ac_system`` fixture makes the name no longer correct. I do think it would be worthwhile to make a second test called ``cec_dc_adr_ac_system`` or something like it. Then you can revert the change to the ``test_dc_models`` fixture, while also updating the ``test_ac_models`` ``ac_systems`` dictionary.

File: pvlib/pvsystem.py [hunk 18]
```diff
@@ -2037,6 +2051,120 @@ def snlinverter(v_dc, p_dc, inverter):
...
+    ======   ============================================================
+    p_nom    The nominal power value used to normalize all power values,
+             typically the DC power needed to produce maximum AC power output,
+             (W).
+
+    v_nom    The nominal DC voltage value used to normalize DC voltage values,
+             typically the level at which the highest efficiency is achieved,
+             (V).
+
+    pac_max  The maximum AC output power value, used to clip the output if
```
Comment (VolkrB) [inline pvlib/pvsystem.py]: should the column header be longer or the variables shorter?

File: pvlib/test/test_modelchain.py (hunk 3 above)
Comment (VolkrB) [inline pvlib/test/test_modelchain.py]: good catch

File: pvlib/pvsystem.py (hunk 18 above)
Comment (wholmgren) [inline pvlib/pvsystem.py]: A longer Column header by one = sign. You'll also need to add one space between all of the Column and the Description sections so that the table stays properly formatted. I guess you should probably also make the Description header longer and/or make the text fit within the existing Description header. 

I'm pretty sure that Sphinx would build the table in its present form and spit out a warning. It's not a big deal if it's just one table, but every silly warning makes it slightly harder to identify other issues with the documentation. If you actually build the docs and don't see a warning then we can ignore all this.

**Comment by VolkrB** [comment]
adrinverter is not being tested in the singlediode dc modelchain, but it's being tested in the ac_model chain. Should I write a new test function for including it in the dc modelchain or maybe there's a better solution?

**Comment by wholmgren** [comment]
No need to write a new test function for the dc modelchain. I'm confused about why the ``'singlediode'`` ``test_dc_models`` test passes, though. It's supposed to check to the second decimal, but you've (I think inadvertently) changed the value in the first decimal.

**Comment by VolkrB** [comment]
failed for check_less_precise = 3, I updated with the correct result

**Comment by wholmgren** [comment]
Thanks. The test errors are unrelated to this PR, so I think we're ready to merge. Agreed?

**Comment by VolkrB** [comment]
Looks good on my end

**Comment by adriesse** [comment]
Agreed.

**Comment by wholmgren** [comment]
thanks @VolkrB and @adriesse!
